hw3/exercise3/src/components/Search.js

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+const url = "https://thronesapi.com/api/v2/Characters";
+function Search() {
+  const [charList, setCharList] = useState([]);
+  const [searchedChar, setSearchedChar] = useState("");
+
+  useEffect(() => {
+    axios.get(url).then((res) => setCharList(res.data));
+  }, []);
+
+  return (
+    <div class="container">
+      <h2>
+        <label for="searchInput">
+          Search for Your Favorite "Game of Thrones" Character:
+        </label>
+      </h2>
+      <input
+        type="text"
+        id="searchInput"
+        className="searchInput"
+        placeholder="Search Character here.."
+        onChange={(e) => setSearchedChar(e.target.value)}
+      />
+
+      <section id="characters" class="mx-auto">
+        {charList
+          .filter((item) => {
+            if (searchedChar === "") {
+              return item;
+            } else if (
+              item.fullName.toUpperCase().includes(searchedChar.toUpperCase())
+            ) {
+              return item;
+            }
+          })
+          .map((item) => {
+            let altText = item.fullName + " of " + item.family;
```

Comment:
According to the Airbnb style guide, it's recommended to use template literals instead of string concatenation when we have variables inside a string.

`let altText = `${item.fullName} of ${item.family}`;
`